src/samplebuffer.cpp

```diff
@@ -0,0 +1,21 @@
+#include "samplebuffer.h"
```

Comment:
I have a slight preference to stick everything in the header file for easy inlining. Is there any reason it needs its own object?